hazelcast/src/test/java/com/hazelcast/multimap/LocalMultiMapStatsTest.java

```diff
@@ -150,9 +155,14 @@ public void testPutAllAndHitsGeneratedKey() {
         Map<Integer, Collection<? extends Integer>> expectedMultiMap = new HashMap<>();
         testPutAllAndHitsGeneratedTemplate(expectedMultiMap,
                 (o) -> {
-                    for (int i = 0; i < 100; ++i) {
-                        o.putAllAsync(i, expectedMultiMap.get(i));
+                    // We need to wait for all putAllAsync() calls to finish
+                    int loopLimit = 100;
+                    CompletableFuture<Void>[] futureList = new CompletableFuture[loopLimit];
+                    for (int i = 0; i < loopLimit; ++i) {
+                        CompletionStage<Void> voidCompletionStage = o.putAllAsync(i, expectedMultiMap.get(i));
+                        futureList[i] = (CompletableFuture<Void>) voidCompletionStage;
```

Comment:
minor: Maybe just this would be enough

```suggestion
                        futureList[i] = o.putAllAsync(i, expectedMultiMap.get(i)).toCompletableFuture();
```

Reply:
I have used toCompletableFuture() instead of casting. Also added comment about the join() call